public/main.js

```diff
@@ -0,0 +1,114 @@
+//ИМПОРТ КЛАССОВ, ПОЧЕМУ ТО, СДЕЛАТЬ НЕ ВЫШЛО. бЫЛО ПЕРЕПРОБЫВАНО ОЧЕНЬ МНОГО. ТАК ЧТО, ИЗВИНИТЕ, НО ОНИ ЗДЕСЬ
+const search_btn = document.getElementById('search_btn');
+const parentBlock = document.getElementById('music_list');
+const libraryStorage = document.getElementById('libraryStorage');
+//ЗАПРОС НА СЕРВЕР
+class HttpRequests {
+    constructor(apiKey, api) {
+        this.requestURL = '';
+        this.apiKey = apiKey;
+        this.api = api;
+    }
+    async sendRequests(url, data) {
+        this.response = await fetch(url, {
+            method: data.method,
+            body: data.body
+        });
+        return await this.response.json();
+    }
+    async searchTrack(track) {
+        this.requestURL = `${this.api}/2.0/?method=track.search&track=${track}&api_key=${this.apiKey}&limit=20&format=json`;
+        return await this.sendRequests(this.requestURL, { method: "GET" });
+    }
+}
+// ОТРИСОВКА ДАННЫХ С СЕРВЕРА
+class Track {
+    constructor() {
+        this.trek = {
+            image: '',
+            treckname: '',
+            author: '',
+        };
+    }
+    addTrack(trek) {
+        this.trek = trek;
+        this.id = `${this.trek.treckname}${this.trek.author}`;
+        return `
+        <div class="content_music_list_item" }>
+            <img src=${this.trek.image} width ="250px" height ="250px">
+            <p class="music_name">Название: ${this.trek.treckname}</br></p>
+            <p class="music_name">Автор: ${this.trek.author}</p>
+            <button type="button" class="add_btn" id='${this.id}'>Добавить</button>
+        </div>
+    `;
+    }
+    dupTrack(trek) {
+        this.trek = trek;
+        this.id = `${this.trek.treckname}${this.trek.author}`;
+        this.deleteId = `${this.trek.treckname}${this.trek.author}${this.trek.author}`;
+        return `
+        <div class="content_music_list_item" id='${this.deleteId}'>
+            <img src=${this.trek.image} width ="250px" height ="250px">
+            <p class="music_name">Название: ${this.trek.treckname}</br></p>
+            <p class="music_name">Автор: ${this.trek.author}</p>
+            <button type="button" class="delete_btn" id='${this.id}'>удалить</button>
+        </div>
+    `;
+    }
+    removeTrack(trek) {
+        this.trek.remove();
+    }
+}
+//ПОЛУЧЕНИЕ ДАННЫХ С СЕРВЕРА И ИХ ОТРИСОВКА
+search_btn.addEventListener('click', async (event) => {
+    parentBlock.innerHTML = "";
+    const api = 'http://ws.audioscrobbler.com';
```

Comment:
Лучше слать запросы через https
```suggestion
    const api = 'https://ws.audioscrobbler.com';
```